File: project10/ruby/Walkingbird.rb
# Walkingbird.rb | Defines the Walkingbird class which inherits attributes and methods
#   from the Bird superclass.
#
# Completed by: Haim Hong
# Date: March 25 2022
####################################################

require './Bird.rb'

class Walkingbird < Bird
    def movement  # movement function for this class specific
        'walked'
    end
end

File: project10/java/WalkingBird.java
/* WalkingBIrd.java is subclass of Bird
 *
 * Completed by: Haim Hong
 * Date: March 25 2022
 ******************************************************/

public class WalkingBird extends Bird{
  /* default constructor
  * PostCond: myName == "".
  */
    public WalkingBird(){
        super();
      }
     /* explicit constructor
  * Receive: name, a String 
  * PostCond: myName == name.
  */
    public WalkingBird(String name){
        super(name);
      }
  /* A WalkingBird's movement
  * Return: a movement for WalkingBird
  */
    public String movement(){
        return "walked";
      }
}